### PR TITLE
Update dockerfiles to support %pip to master

### PR DESCRIPTION
**Pull request by deka108**
This PR updates Dockerfile to match how utilities related to Python environments are installed in DBR to support Python library features use cases.

We also updated the existing installed python package version on the file to match the latest DBR version: https://docs.databricks.com/release-notes/runtime/11.1.html#system-environment

### CODE REVIEW

File: ubuntu/python/Dockerfile [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
-  pyarrow==4.0.0 \
-  matplotlib==3.4.2 \
+  ipython==7.32.0 \
+  numpy==1.20.3 \
+  pandas==1.3.4 \
+  pyarrow==7.0.0 \
+  matplotlib==3.4.3 \
   jinja2==2.11.3 \
-  ipykernel==6.21.1
+  ipykernel==5.3.4
```
Comment (vperiyasamy) [inline ubuntu/python/Dockerfile:38]: any reason to roll back the ipykernel version?

Comment (deka108) [reply ubuntu/python/Dockerfile:38]: ah good catch this should be 6.12.1!

Comment (deka108) [reply ubuntu/python/Dockerfile:38]: Interestingly the original Dockerfile is a bit wrong too as it should be `6.12.1` instead of `6.21.1`

Comment (xinzhao-db) [reply ubuntu/python/Dockerfile]: wait, this is a intended fix by https://github.com/databricks/containers/commit/d34cdcfeb3719f6ae659a3a8653baf94e5a27a1b

Comment (vperiyasamy) [reply ubuntu/python/Dockerfile]: @xinzhao-db can you provide more context? didn't see any description in that PR

Comment (xinzhao-db) [reply ubuntu/python/Dockerfile]: Confirmed with @xiaochen-db , the correct version is 6.12.1

**Review by xinzhao-db — APPROVED** [review]

**Review by vperiyasamy — APPROVED** [review]

**Pull request merged**